YAXLib/YAXSerializer.cs

```diff
@@ -2833,7 +2833,7 @@ private IEnumerable<MemberWrapper> GetFieldsToBeSerialized(UdtWrapper typeWrappe
         /// <returns>the sequence of fields to be serialized for the serializer's underlying type.</returns>
         private IEnumerable<MemberWrapper> GetFieldsToBeSerialized()
         {
-            return GetFieldsToBeSerialized(m_udtWrapper);
+            return GetFieldsToBeSerialized(m_udtWrapper).OrderBy(t=>t.Order);

```

Comment:
https://msdn.microsoft.com/en-us/library/bb549422%28v=vs.100%29.aspx

"This method performs a stable sort; that is, if the keys of two elements are equal, the order of the elements is preserved."